src/domain/locking/entities/schemas/locking-event.schema.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  LockEvent,
+  LockType,
+  LockingEvent,
+  UnlockEvent,
+  WithdrawEvent,
+} from '@/domain/locking/entities/locking-event.entity';
+import { JSONSchemaType } from 'ajv';
+import { Page } from '@/routes/common/entities/page.entity';
+
+const LOCK_EVENT_SCHEMA_ID =
+  'https://safe-client.safe.global/schemas/locking/lock-event.json';
+
+const lockEventSchema: JSONSchemaType<LockEvent> = {
+  $id: LOCK_EVENT_SCHEMA_ID,
+  type: 'object',
+  properties: {
+    type: { type: 'string', const: LockType.LOCK },
+    amount: { type: 'string' },
+    executedAt: { type: 'string' },
+  },
+  required: ['type', 'amount', 'executedAt'],
+};
+
+const UNLOCK_EVENT_SCHEMA_ID =
+  'https://safe-client.safe.global/schemas/locking/unlock-event.json';
+
+const unlockEventSchema: JSONSchemaType<UnlockEvent> = {
+  $id: UNLOCK_EVENT_SCHEMA_ID,
+  type: 'object',
+  properties: {
+    type: { type: 'string', const: LockType.UNLOCK },
+    amount: { type: 'string' },
+    executedAt: { type: 'string' },
+    unlockIndex: { type: 'string' },
+    unlockedAt: { type: 'string' },
+  },
+  required: ['type', 'amount', 'executedAt', 'unlockIndex', 'unlockedAt'],
+};
+
+const WITHDRAW_EVENT_SCHEMA_ID =
+  'https://safe-client.safe.global/schemas/locking/withdraw-event.json';
+
+const withdrawEventSchema: JSONSchemaType<WithdrawEvent> = {
+  $id: WITHDRAW_EVENT_SCHEMA_ID,
+  type: 'object',
+  properties: {
+    type: { type: 'string', const: LockType.WITHDRAW },
+    amount: { type: 'string' },
+    executedAt: { type: 'string' },
+    unlockIndex: { type: 'string' },
+  },
+  required: ['type', 'amount', 'executedAt', 'unlockIndex'],
+};
+
+const LOCKING_EVENT_SCHEMA_ID =
+  'https://safe-client.safe.global/schemas/locking/locking-event.json';
+
+const lockingEventSchema: JSONSchemaType<LockingEvent> = {
+  $id: LOCKING_EVENT_SCHEMA_ID,
+  type: 'object',
+  discriminator: { propertyName: 'type' },
+  oneOf: [lockEventSchema, unlockEventSchema, withdrawEventSchema],
+};
+
+export const LOCKING_EVENT_PAGE_SCHEMA_ID =
+  'https://safe-client.safe.global/schemas/locking/locking-event-page.json';
+
+// TODO: Use buildPageSchema when types are fixed
```

Comment:
One of the main differences that I see between `buildPageSchema` and `lockingEventPageSchema` is that `count` is nullable in `lockingEventPageSchema`. But besides that, `buildPageSchema` can be used or?

Reply:
`count` [is nullable everywhere](https://github.com/safe-global/safe-client-gateway/pull/1192), afaiu. Whilst we can use the `buildPageSchema` function, we'd need to coerce the type to `Schema`, which is inherently wrong.

I did manage to get the types working but it requires changing a lot of schemas across the project. I would sooner leave this comment and create a function with `zod` when we've migrated. Would you prefer I do so in this PR though?

Reply:
> Would you prefer I do so in this PR though?

No let's tackle these issues in the Zod migration. This was more so that the schemas would be aligned across the project (as we already have one schema for the `Page`).